integration-tests/tracing/src/test/java/io/kiota/quarkus/it/QuarkusKiotaResourceTest.java

```diff
@@ -0,0 +1,57 @@
+package io.kiota.quarkus.it;
+
+import static io.restassured.RestAssured.get;
+import static io.restassured.RestAssured.given;
+import static java.net.HttpURLConnection.HTTP_OK;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.common.mapper.TypeRef;
+
+@QuarkusTest
+public class QuarkusKiotaResourceTest {
+
+    @BeforeEach
+    @AfterEach
+    void reset() {
+        given().get("/reset").then().statusCode(HTTP_OK);
+        await().atMost(5, SECONDS).until(() -> getSpans().size() == 0);
+    }
+
+    private List<Map<String, Object>> getSpans() {
+        return get("/export").body().as(new TypeRef<>() {
+        });
+    }
+
+    @Test
+    public void testHelloEndpoint() {
+        given().when()
+                .get("/quarkus-kiota2")
+                .then()
+                .statusCode(200)
+                .body(is("{\"value\":\"Hello quarkus-kiota myself\"}"));
+
+        await().atMost(Duration.ofMinutes(2)).until(() -> getSpans().size() > 0);
+        Map<String, Object> spanData = getSpans().stream().filter(span -> span.get("kind").equals("CLIENT")).findAny().get();
```

Comment:
Shouldn't you be getting 2 spans in a parent, child relationship?

Reply:
We are receiving more spans here because this test involves:

- Restassured invoking a Quarkus endpoint
- The Quarkus server spins up a VertX client and invokes another endpoint
- The Quarkus server returns the first answer to itself
- The Quarkus server returns to the Restassured client

Would you suggest different assertions?

Reply:
yup, I can add a few assertions on that too 👍

Reply:
done, let me know if you think this is enough 🙂